### PR TITLE
Some updates to the template

### PR DESCRIPTION
**Pull request by baruchiro**

### CODE REVIEW

File: template/typescript/.editorconfig [hunk 1]
```diff
@@ -1,9 +0,0 @@
-root = true
-
-[*]
-charset = utf-8
-indent_style = space
-indent_size = 2
-end_of_line = lf
-insert_final_newline = true
-trim_trailing_whitespace = true
```
Comment (baruchiro) [inline template/typescript/.editorconfig:9]: This is a user-specific file, we need to remove it from template

File: template/typescript/package.json [hunk 2]
```diff
@@ -7,15 +7,19 @@
   "repository": "{{repo}}",
   "main": "dist/index.js",
   "module": "dist/index.es.js",
+  "types": "dist/index.d.ts",
   "jsnext:main": "dist/index.es.js",
+  "files": [
+    "dist"
+  ],
```
Comment (baruchiro) [inline template/typescript/package.json]: Currently, you're publishing the whole code, but we need to publish only the output files.
Here I set the `dist` to be outputted (alongside the default `package.json` and `README.md`).
In the `.npmignore` I removed the `*.test.*` files from the package.

File: template/typescript/package.json [hunk 3]
```diff
@@ -7,15 +7,19 @@
...
   "engines": {
     "node": ">=8",
     "npm": ">=5"
   },
   "scripts": {
     "test": "cross-env CI=1 react-scripts test --env=jsdom",
     "test:watch": "react-scripts test --env=jsdom",
-    "build": "rollup -c",
+    "build": "rollup -c && tsc -d --emitDeclarationOnly --noEmit false --declarationDir dist",
```
Comment (baruchiro) [inline template/typescript/package.json:19]: A workaround for `@rollup/plugin-typescript`, see https://github.com/rollup/plugins/issues/287#issuecomment-650581821

File: template/typescript/tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,5 @@
 {
   "compilerOptions": {
-    "outDir": "build",
```
Comment (baruchiro) [inline template/typescript/tsconfig.json:3]: Part of the above workaround

File: template/typescript/package.json (hunk 2 above)
Comment (Hermanya) [reply template/typescript/package.json]: As I described in https://github.com/Hermanya/create-react-hook/issues/22 the problem here is that when I publish create react hook, only the first folder will upload to npm. I'm not sure how to differentiate between the npm publish I do and the npm publish the users of this tool will run on packages created.

Comment (baruchiro) [reply template/typescript/package.json]: OK, I removed this.

**Comment by Hermanya** [comment]
thank you for your contribution! this is now available in v3.1.0

**Pull request merged**